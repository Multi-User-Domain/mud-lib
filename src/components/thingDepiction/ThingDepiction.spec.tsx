import React from "react";
import { render } from "@testing-library/react";
import { createThing, addUrl } from "@inrupt/solid-client";
import { FOAF } from "@inrupt/lit-generated-vocab-common";
import { ThingDepiction } from "./ThingDepiction";

const mockUrl = "http://test.url/image.png";
const mockThingWithDepiction = addUrl(createThing(), FOAF.depiction, mockUrl);

it("renders image when present", () => {
  const { asFragment } = render(
    <ThingDepiction thing={mockThingWithDepiction} />
  );
  expect(asFragment()).toMatchSnapshot();
});

const mockThingWithoutDepiction = createThing();

it("renders placeholder when no image present", () => {
  const { asFragment } = render(
    <ThingDepiction thing={mockThingWithoutDepiction} />
  );
  expect(asFragment()).toMatchSnapshot();
});
