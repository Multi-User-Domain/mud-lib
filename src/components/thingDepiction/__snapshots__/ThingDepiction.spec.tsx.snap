// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders image when present 1`] = `
<DocumentFragment>
  <img
    class="chakra-image__placeholder css-0"
  />
</DocumentFragment>
`;

exports[`renders placeholder when no image present 1`] = `
<DocumentFragment>
  <div
    class="css-tkevr6"
  >
    <div
      class="css-1on1w5z"
    />
  </div>
</DocumentFragment>
`;
